tests/templates/test_state_preparations.py

```diff
@@ -359,6 +389,98 @@ def circuit(state_vector):
             state_vector = np.array([0, 2, 0, 0])
             circuit(state_vector)
 
+    # fmt: off
+    @pytest.mark.parametrize("state_vector, n_wires", [
+        ([.5,.5,.5,.5],2),
+        ([1, 0, 0, 0], 2),
+        ([0, 1, 0, 0], 2),
+        ([0, 0, 0, 1], 2),
+        ([0, 1, 0, 0, 0, 0, 0, 0], 3),
+        ([0, 0, 0, 0, 1, 0, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+        ([1/2, 0, 0, 0, 1/2, 1/2, 1/2, 0], 3),
+        ([1/3, 0, 0, 0, 2/3, 2/3, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+    ])
+    # fmt: on
+
+    def test_RZ_skipped(self, state_vector, n_wires):
+        """Tests whether the cascade of RZ gates is skipped.
+        Note that currently there are also inefficiencies in the number of CNOT gates in the RY cascade.
+        When these are updated, this test will become ineffective"""
```

Comment:
I'm curious where the inefficiencies come from with the RY cascade? Is this something we could fix in this PR?

Reply:
The RY cascade behaves similarly to the RZ cascade. When the RY rotations are 0, the CNOT gates are sometimes still applied. We can change it in this PR.

Reply:
Oh I see. When does it happen that (some of?) the `RY` rotations are 0? Is this the case where some elements of the distribution have zero value?

Reply:
Exactly. E.g. to prepare the state `|100>` we only need to apply `RY` to qubit `0`. But we end up with several `CNOT` gates.